### ISSUE TITLE
Incorrect (too large) sample count seen in BAM output - possibly triggered by default trimming

### ISSUE THREAD

**Issue by dkj** [nanoporetech/dorado#520]
I'm having difficulty reconsiling `mv:` [BAM auxtag] data with the `ns:` and `ts:`. Looking into this I found BAM records where the number of samples `ns:` was greater than the `sample_count` in the original pod5 data. `ns:` in BAM should always be the same as `sample_count` from pod5, right? Never larger anyway I think... so, bug?

Reproducer from your public data set at `s3://ont-open-data/giab_2023.05/` :

Original sample count
```
$ pod5 inspect read PAO83395_pass__ba06e1bb_99fc9efe_539.pod5 5178a22c-fbcf-448f-b642-587138eb64c2 | grep sample_count
	sample_count: 500390
```

From default dorado 0.5.0 default sup basecall, see increase in reported number of samples :
``` 
$ dorado basecaller sup --emit-moves PAO83395_pass__ba06e1bb_99fc9efe_539.pod5 > PAO83395_pass__ba06e1bb_99fc9efe_539.noalign.sup.bam
[2023-12-12 14:45:42.667] [info]  - downloading dna_r10.4.1_e8.2_400bps_sup@v4.3.0 with httplib
[2023-12-12 14:45:43.848] [info] > Creating basecall pipeline
[2023-12-12 14:45:55.412] [info]  - set batch size for cuda:0 to 1280
[2023-12-12 14:46:16.851] [info] > Simplex reads basecalled: 3999
[2023-12-12 14:46:16.851] [info] > Simplex reads filtered: 1
[2023-12-12 14:46:16.851] [info] > Basecalled @ Samples/s: 6.520208e+06
[2023-12-12 14:46:16.873] [info] > Finished

$ samtools view -N <(echo 5178a22c-fbcf-448f-b642-587138eb64c2) PAO83395_pass__ba06e1bb_99fc9efe_539.noalign.sup.bam | perl -F"\t" -nale 'print join"\t",$F[0],(grep{m{^[tn]s:}}@F)'
5178a22c-fbcf-448f-b642-587138eb64c2	ns:i:500996	ts:i:616
```

But get the correct number of samples if trimming is turned off:
```
$ dorado basecaller sup --emit-moves --no-trim PAO83395_pass__ba06e1bb_99fc9efe_539.pod5 > PAO83395_pass__ba06e1bb_99fc9efe_539.noalign.notrim.sup.bam
[2023-12-12 15:27:20.359] [info]  - downloading dna_r10.4.1_e8.2_400bps_sup@v4.3.0 with httplib
[2023-12-12 15:27:21.607] [info] > Creating basecall pipeline
[2023-12-12 15:27:33.132] [info]  - set batch size for cuda:0 to 1600
[2023-12-12 15:27:55.666] [info] > Simplex reads basecalled: 3999
[2023-12-12 15:27:55.666] [info] > Simplex reads filtered: 1
[2023-12-12 15:27:55.666] [info] > Basecalled @ Samples/s: 6.202622e+06
[2023-12-12 15:27:55.690] [info] > Finished

$ samtools view -N <(echo 5178a22c-fbcf-448f-b642-587138eb64c2) PAO83395_pass__ba06e1bb_99fc9efe_539.noalign.notrim.sup.bam | perl -F"\t" -nale 'print join"\t",$F[0],(grep{m{^[tn]s:}}@F)'
5178a22c-fbcf-448f-b642-587138eb64c2	ns:i:500390	ts:i:10
```

**Comment by tijyojwad** [comment]
Hi @dkj - thank you so much for raising this! yes definitely a bug.

**Comment by tijyojwad** [comment]
Hi @dkj - this should now be fixed with release v0.5.1 - https://github.com/nanoporetech/dorado#installation
